src/rgw/rgw_lua_background.h

```diff
@@ -0,0 +1,77 @@
+  #pragma once
+  #include "common/dout.h"
+  #include "rgw_common.h"
+  #include <string>
+  #include "rgw_lua_utils.h"
+
+  namespace rgw::lua {
+
+  //Interval between each execution of the script
+  //is set to 5 seconds, this setting is configurable
+  //using RGW.setoff from within the lua script
+  constexpr const int INIT_SET_OFF = 5;
+ 
+  static std::mutex m_mtx;
+  //Writeable meta table named RGW with mutex protection
+  using BackgroundMap = std::unordered_map<std::string, std::string>;
+  struct RGWTable : StringMapMetaTable<BackgroundMap,
+    StringMapWriteableNewIndex<BackgroundMap>> {
+      static std::string TableName() {return "RGW";}
+      static std::string Name() {return TableName() + "Meta";}
+      static int IndexClosure(lua_State* L) {
+          std::lock_guard l(m_mtx);
+          return StringMapMetaTable::IndexClosure(L);
+      }
+      static int PairsClosure(lua_State* L) {
```

Comment:
i think we can remove the "pairs" interface
don't thin kit would make sense to iterate over the RGW table
also, not sure about thread safety of the iterator